internal/services/servicefabricmanaged/service_fabric_managed_cluster_resource.go

```diff
@@ -789,7 +789,7 @@ func expandClusterProperties(model *ClusterResourceModel) *managedcluster.Manage
 				DestinationAddressPrefixes: &[]string{"0.0.0.0/0"},
 				Direction:                  managedcluster.DirectionInbound,
 				Name:                       fmt.Sprintf("rule%d-allow-fe", rule.FrontendPort),
-				Priority:                   1000,
+				Priority:                   1000 + int64(idx),
```

Comment:
instead of doing this should we maybe just expose the priority so the user can set it instead of making an assumption for them?

Reply:
@katbyte  Sure, thanks for your time. I would like to confirm that only the `Priority` of NetworkSecurityRule needs to be exposed? Currently, all of properties of NetworkSecurityRule are hardcoded.

Reply:
is there a reason for that and not exposing them to a user?

Reply:
Thank you for your reply. Sorry for the late response. I don't know why they are not exposed to the user. This resource was contributed by koikonom in [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/14131). Maybe he has more knowledge in this field.

Reply:
well regardless we should expose this one? as well name and direction? these seem to make sense?

they will have to be computed

Reply:
> well regardless we should expose this one? as well name and direction? these seem to make sense?
> 
> they will have to be computed

I don't quite understand what you mean, could you explain?

Reply:
could we expose these properties in schema, ie add priority with a default of 1000m and computed.

Reply:
@katbyte thank you for your reply. Go through the [NSG rules guidance](https://docs.microsoft.com/en-us/azure/service-fabric/how-to-managed-cluster-networking#nsg-rules-guidance) and [NetworkSecurityRule ](https://docs.microsoft.com/en-us/azure/templates/microsoft.servicefabric/managedclusters?tabs=bicep#networksecurityrule), I assume that it makes more sense to expose all the properties instead of some properties of NSRules to user, WDYT? I have updated the code, could you give me some suggestions?